### ISSUE TITLE
Zuul proxies do not load balance routes correctly

### ISSUE THREAD

**Issue by jdubois** [jhipster/generator-jhipster#5664]
This was working perfectly fine on older releases of JHipster, but it looks like Spring Cloud modified the way they do load balancing (and ping) for Zuul routes.
Please note this is only for the Zuul routes that are automatically created when a service is registered in Eureka, if routes are configured manually it's an even harder issue (but that's another issue).
From what I experience:

- The Zuul route always uses the first instance, and not the others
- When the instance goes down, after the Gateway gets synced with Eureka, the route "quickly" uses another instance

So you have no "real" load balancing, and downtime when an instance gets killed.

**Comment by stevehouel** [comment]
@jdubois If you want we can work on it together. You can also use my code which test the loadbalancing (the one you show on Devoxx). You can find it on https://github.com/stevehouel/jhipster-story

**Comment by jdubois** [comment]
@stevehouel do you have time to work on this today?

My current experiment is by doing:


    @Configuration
    public static class RibbonConfiguration {

        @Bean
        public IRule ribbonRule(IClientConfig config) {
            WeightedResponseTimeRule rule = new WeightedResponseTimeRule();
            rule.initWithNiwsConfig(config);
            return rule;
        }

        @Bean
        public IPing ribbonPing() {
            return new NIWSDiscoveryPing();
        }
    }

But it doesn't really work at the moment :-(

**Comment by stevehouel** [comment]
@jdubois I can test it today if you want, I will update my jhipster-story with this configuration and deploy it (maybe on my rancher cluster). 

What I understand is that you want to use the DiscoveryPing and not the default Ribbon Eureka impl. I will have a look on algo for the WeightResponseTimeRule but it looks better than default RoundRobinRule

**Comment by stevehouel** [comment]
Another way for LB rule is BestAvailableRule or AvailabilityFilteringRule

**Comment by jdubois** [comment]
@stevehouel from my understanding, the current implementations are just `ZoneAvoidanceRule` and `NoOpPing` see http://cloud.spring.io/spring-cloud-netflix/spring-cloud-netflix.html - and that would explain why it doesn't work anymore

**Comment by stevehouel** [comment]
Yes I agress, NoOpPing disable check of each server using ping request (which was the one used before the update). Now it's only based on eureka server list and status. ZoneAvoidanceRule is for big cloud infrastructure with many Availability Zone. But for us this will not make any distinction between our instances.

**Comment by jdubois** [comment]
- According to http://cloud.spring.io/spring-cloud-netflix/spring-cloud-netflix.html this should in fact work if you configure `eureka.instance.metadataMap.zone`. So I configured this -> I'm going to commit my patch and link it here
- I got this working fine on my laptop, if I run everything directly: I debugged it and both the load balancer and ping are OK, and when I kill an instance (out of 2), I see the load balancer working for a small amount of time (one request out of 2 is failing), and then when the ping sees the instance is down everything is OK (it fully switches to the running instance)
- But on Docker it still seems to be failing -> I have the behavior I described in my first comment, with the load balancer and ping not doing their jobs correctly. @stevehouel will you be able to test it? Maybe I did something wrong
